l10n_be_cooperator_national_number/models/subscription_request.py

```diff
@@ -25,19 +31,38 @@ def _compute_require_national_number(self):
             self.company_id.require_national_number and not self.is_company
         )
 
-    def get_national_number_from_partner(self, partner):
-        national_number_id_category = self.env.ref(
+    @api.model
+    def _get_be_national_register_number_id_category(self):
+        return self.env.ref(
             "l10n_be_partner_identification.l10n_be_national_registry_number_category"
-        ).id
+        )
+
+    @api.model
+    def get_national_number_from_partner(self, partner):
+        national_number_id_category = (
+            self._get_be_national_register_number_id_category()
+        )
         national_number = partner.id_numbers.filtered(
-            lambda rec: rec.category_id.id == national_number_id_category
+            lambda rec: rec.category_id.id == national_number_id_category.id
         )
         return national_number.name
 
+    @api.model
+    def check_be_national_register_number(self, national_number):
+        national_number_id_category = (
+            self._get_be_national_register_number_id_category()
+        )
+        # this function checks the value of id_number.name, not id_number
+        # directly.
+        id_number = namedtuple("id_number", ("name"))(national_number)
+        national_number_id_category.validate_id_number(id_number)
```

Comment:
Nice hack.

I'm a touch ambivalent about the rest of the control flow. `_get_be_national_register_number_id_category` should never return an empty record, which is good. But is it possible that the national number passed into this function is not a Belgian one? If yes, this method might still be fine, but the control flow elsewhere would need to change.

Reply:
this module is belgium-specific, but i think however that the name of the `national_number` field should be belgium-specific too. anyway, i don’t think that `cooperator` currently supports multiple localization modules at the same time (although it might work).